modules/packages/ArgumentParser.chpl

```diff
@@ -425,19 +507,69 @@ module ArgumentParser {
     }
  }
 
+  /*
+  A parser that performs the following functions:
+
+    1. Stores definitions of valid arguments/flags/options
+    2. Parses an array of arguments as passed from the command line
+       (where args[0] is typically the name of the executable)
+    3. Throws an ArgumentError if invalid or unrecognized arguments are encountered
+
+  The parser can process command line arguments that include or omit an equals
+  sign ``=``, with short ``-`` or long ``--`` indicators.
+
+  We can declare a new argumentParser like so:
+
+  .. code-block:: chapel
+
+    var parser = new argumentParser();
+
+  **Limitations**
+
+   * Cannot place short flags together. That is, if a parser has defined short
+     flags ``-c``, ``-a`` and ``-t``, we cannot handle ``-cat``.
+
+   * Short options must be separated from their values with an equals sign or
+     spaces. That is, ``-cValue`` is not supported and should be entered as either
+     ``-c Value`` or ``-c=Value``.
+
+   * Numeric types are not supported and will be passed as strings.
+
+   * Negative numbers on the command line may cause issues.
```

Comment:
Because e.g. `-1` might be thought of as an option? Is it a problem with an argument?
I would expect it to work with arguments & options with arguments. (And if not I'd view it as a bug).

Reply:
You're right, this should work. I had made a bad decision early on to check for leading `-` in the `_match` routine for options, but now I see that we should just consume it and let the dev/user sort it out if it's a bad value or not. 
I updated the parser and the test for that condition, plus added some tests showing that arguments and options both work with negative numbers.

Reply:
That'll be another PR, right? Can you link it from here when you make it?

Reply:
A bugfix was put into place for handling negative numbers in PR #18395